tools/cxbe/Xbe.cpp

```diff
@@ -747,6 +748,32 @@ Xbe::Xbe(class Exe *x_Exe, const char *x_szTitle, bool x_bRetail, const std::vec
                 szBuffer += sizeof(*m_LibraryVersion);
                 hwc += sizeof(*m_LibraryVersion);
             }
+
+            m_KernelLibraryVersion = new LibraryVersion;
```

Comment:
* `XBOXKRNL` and `XAPILIB` library version decls should be counted in `m_Header.dwLibraryVersions`
* Should eliminate the dummy CXBEx lib version if we are populating these ones
* Might be simpler to just increase the count, let the above code do the allocation, then fill in the info after

Reply:
> `XBOXKRNL` and `XAPILIB` library version decls should be counted in m_Header.dwLibraryVersions
> [...]
> Might be simpler to just increase the count, let the above code do the allocation, then fill in the info after

Did you check on MS XBEs? They differentiate between kernel and non-kernel libs (I didn't remember they also track XAPI separately), so I'm not sure if these 2 count.

> Should eliminate the dummy CXBEx lib version if we are populating these ones

I believe VerifierX still needs at least one non-kernel lib?
XDK always has at least one valid lib, because of D3D etc.

Reply:
> Did you check on MS XBEs

Yes, here's Halo. These version pointers in the header simply point into the array of all library versions. Maybe there is an exception in some other XBEs, but this is how I've always seen it.

```python
In [3]: xbe = Xbe.from_file("default.xbe")

In [4]: xbe.header.lib_versions_count
Out[4]: 8

In [5]: xbe.libraries
Out[5]: 
{'XAPILIB': <XbeLibrary 'XAPILIB' (1.0.3911)>,
 'D3D8': <XbeLibrary 'D3D8' (1.0.3925)>,
 'D3DX8': <XbeLibrary 'D3DX8' (1.0.3911)>,
 'DSOUND': <XbeLibrary 'DSOUND' (1.0.3936)>,
 'DSOUNDH': <XbeLibrary 'DSOUNDH' (1.0.3937)>,
 'XNETS': <XbeLibrary 'XNETS' (1.0.3911)>,
 'XBOXKRNL': <XbeLibrary 'XBOXKRNL' (1.0.3911)>,
 'LIBC': <XbeLibrary 'LIBC' (1.0.3911)>}
```